plugins/modules/get_certificate.py

```diff
@@ -381,6 +413,39 @@ def main():
                 ciphers_joined = ":".join(ciphers)
                 ctx.set_ciphers(ciphers_joined)
 
+            if tls_ctx_options is not None:
+                # Clear default ctx options
+                ctx.options = 0
+
+                # For each item in the tls_ctx_options list
+                for tls_ctx_option in tls_ctx_options:
+                    # If the item is a string_type
+                    if isinstance(tls_ctx_option, string_types):
+                        # Convert tls_ctx_option to a native string
+                        tls_ctx_option_str = to_native(tls_ctx_option)
+                        # Get the tls_ctx_option_str attribute from ssl
+                        tls_ctx_option_attr = getattr(ssl, tls_ctx_option_str)
```

Comment:
```suggestion
                        tls_ctx_option_attr = getattr(ssl, tls_ctx_option_str, None)
```
Otherwise you end up with an uncaught exception if a random string is provided that doesn't exist as an attribute of `ssl`.

Reply:
Random strings are actually handled as-is. I tried adding `OP_ALL_DOES_NOT_EXIST` to the options list without adding `, None` to `getattr` and this is the output:
```
TASK [Get Certificate] ********************************************************************************************************************************
fatal: [REDACTED]: FAILED! => {"changed": false, "msg": "Failed to get cert from REDACTED, error: module 'ssl' has no attribute 'OP_ALL_DOES_NOT_EXIST'"}
```

Do you still want this suggestion implemented?

Reply:
The global try/catch is for catching errors in the actual certificate retrieval, not for argument handling. The error message also contains information on a proxy (if configured), which would be very confusing in this case. I think it's better to have clear error handling here which is not related ot failed certificate retrieval.